frontend/package.json

```diff
@@ -40,7 +40,9 @@
 		"sass-loader": "^8.0.2",
 		"typescript": "^4.4.4",
 		"vue-loader": "^16.8.2",
-		"vue-template-compiler": "^2.6.14",
+		"vue-template-compiler": "^2.6.14"
```

Comment:
Let's save this change for another PR, since it might have some unintended consequences. Such as: the change to `package-lock.json`, which might or might not be related to this change, has had 15.5k lines added to it for some reason. We should investigate this further. But for now, please discard the changes to `package-lock.json` and `package.json` right before we merge this PR.

Reply:
I went ahead and made #395 which, with some careful experimentation, upgrades everything that we can possibly upgrade except the things that are blocked on Vue CLI 5.0 which is in the release candidate stage right now. This also cleaned up some new linter errors and things. Some of this might be a duplication of your work, but it was better suited for another PR. If you made any improvements that I didn't include also in that commit, please feel free to include them. Sorry if this means you have some merge conflicts, I am more than happy to help you sort them out.